packages/react-components/react-utilities/src/hooks/useMotionPresence.ts

```diff
@@ -0,0 +1,345 @@
+import * as React from 'react';
+import { useTimeout } from './useTimeout';
+
+/**
+ * CSS Typed Object Model
+ * @see https://drafts.css-houdini.org/css-typed-om-1/
+ * @see https://developer.mozilla.org/en-US/docs/Web/API/CSSUnitValue
+ */
+interface CSSUnitValue {
+  value: number;
+  readonly unit: string;
+}
+
+/**
+ * Style property map read only.
+ * @see https://developer.mozilla.org/en-US/docs/Web/API/StylePropertyMapReadOnly
+ */
+interface StylePropertyMapReadOnly {
+  [Symbol.iterator](): IterableIterator<[string, CSSUnitValue[]]>;
+
+  get(property: string): CSSUnitValue | undefined;
+  getAll(property: string): CSSUnitValue[];
+  has(property: string): boolean;
+  readonly size: number;
+}
+
+/**
+ * HTMLElement with styled map.
+ * @see https://developer.mozilla.org/en-US/docs/Web/API/Element/computedStyleMap
+ */
+type HTMLElementWithStyledMap<TElement extends HTMLElement = HTMLElement> = TElement & {
+  computedStyleMap(): StylePropertyMapReadOnly;
+};
+
+interface CSSWithNumber {
+  number(value: number): {
+    value: number;
+    readonly unit: string;
+  };
+}
```

Comment:
```suggestion
declare global {
  /**
   * CSS Typed Object Model
   * @see https://drafts.css-houdini.org/css-typed-om-1/
   * @see https://developer.mozilla.org/en-US/docs/Web/API/CSSUnitValue
   */
  interface CSSUnitValue {
    value: number;
    readonly unit: string;
  }

  /**
   * Style property map read only.
   * @see https://developer.mozilla.org/en-US/docs/Web/API/StylePropertyMapReadOnly
   */
  interface StylePropertyMapReadOnly {
    [Symbol.iterator](): IterableIterator<[string, CSSUnitValue[]]>;

    get(property: string): CSSUnitValue | undefined;
    getAll(property: string): CSSUnitValue[];
    has(property: string): boolean;
    readonly size: number;
  }

  /**
   * HTMLElement with styled map.
   * @see https://developer.mozilla.org/en-US/docs/Web/API/Element/computedStyleMap
   */
  interface HTMLElement {
    computedStyleMap(): StylePropertyMapReadOnly;
  }

  namespace CSS {
    export function number(value: number): CSSUnitValue;
  }
}
```

Seems like you're providing types that already exist in global scope, but aren't available in our current support matrix.

Since those types are already defined in the global scope anyway, how about we just provide them again in the global scope with a declaration statement?!